### PR TITLE
Fixes #25836: Remove AIX password hash support

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/25836

I tested the current user management technique (with a `<AUTOSUBVARIABLES>AIX</AUTOSUBVARIABLES>`) and it works well.

This unlocks upgrade of `commons-codec-version` (https://github.com/Normation/rudder/pull/5976#discussion_r1827741654).

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala [hunk 1]
```diff
@@ -189,28 +158,13 @@ object HashAlgoConstraint extends Enum[HashAlgoConstraint] {
...
     case object Linux extends DerivedPasswordType {
       final val name = "Unix"
 
       def hash(h: HashAlgoConstraint): HashAlgoConstraint = h match {
-        case AixMD5    => LinuxShadowMD5
-        case AixSHA256 => LinuxShadowSHA256
-        case AixSHA512 => LinuxShadowSHA512
-        case x         => x
+        case x => x
       }
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala:166]: The pattern matching is useless, you can remove it completely

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala [hunk 2]
```diff
@@ -1386,8 +1385,7 @@ object BuildNodeConfiguration extends Loggable {
                                                                                  case (k, v) =>
                                                                                    // js lib is specific to the node os, bind here to not leak eval between vars
                                                                                    val jsLib = context.nodeInfo.os.os match {
-                                                                                     case AixOS => JsRudderLibBinding.Aix
-                                                                                     case _     => JsRudderLibBinding.Crypt
+                                                                                     case _ => JsRudderLibBinding.Crypt
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala:1388]: Same, you can remove the pattern matching

**Review by fanf — COMMENTED** [review]
Lgtm appart the little changed, youbcan self merge after changing them

**Comment by amousset** [comment]
PR updated with a new commit

**Pull request merged**